lsp/nls/src/analysis.rs

```diff
@@ -134,6 +134,42 @@ impl<'a> ParentChainIter<'a> {
         }
     }
 
+    /// Like `next`, but skips over everything except for merges, annotations, and records.
+    pub fn next_merge(&mut self) -> Option<RichTerm> {
+        let is_fieldy_term = |rt: &RichTerm| {
+            matches!(
+                rt.as_ref(),
+                Term::Op2(BinaryOp::Merge(_), _, _)
```

Comment:
It's probably not important for the LSP, but FYI theoretically there is also the `MergeContract` operator in `NAryOp`, which is the variant of merge used for contract application. That being said, it's only generated at runtime from `ApplyContract`, which themselves come from annotations, which you handle.

Reply:
Thanks for the info! I guess for now I'll just make a note of this in a comment.